Un hombre en un bote de remos situado en P (ver figura) a 5 millas en línea recta del punto A más cercano a una costa, desea llegar al punto B, a 6 millas de A a lo largo de la costa, en el tiempo más corto. ¿A qué distancia del punto A debería desembarcar si puede remar a 2 millas por hora y caminar a 4 millas por hora? Emplee el método de la secante con una precisión de 10−7. Use 15 decimales.

![Imagen del poste](../images/poste.png)

### Análisis del problema

Dado que la pesona va a remar en la trayectoria de *P* a *C*, donde sabemos que *C* es desconocido, ponemos llamar a esta distancia *h*, esta trayectoria *h* tomando en cuenta que es una hipotenusa vendria dada por pitágoras, si denominamos a la distancia entre *A* y *C* como *x*, tendríamos que:

$$ h = \sqrt{5^2 + x^2}  $$

El tiempo que le tomará es igual a la distancia entre la velocidad

$$ tiempo_r = \frac{espacio}{velocidad} = \frac{\sqrt{5^2 + x^2}}{2 m.h } $$

Si tenemos *x* para encontrar la distancia de *C* a *B* seria *6 - x*, por tanto la función es:

$$ tiempo_p = \frac{espacio}{velocidad} = \frac{6-x}{4 m.h } $$

Por lo tanto nuestra función total a optimizar es: 

$$ tiempo =  \frac{\sqrt{25 + x^2}}{2} +  \frac{6-x}{4} $$

Dado que necesitamos el menor tiempo posible, lo que debemos hacer es derivar la función, ya que en el punto en que la pendiente de nuestra función sea cero, tendremos el menor tiempo posible

$$ F_x = \frac{x}{2 \sqrt{25 + x^2}} - \frac{1}{4} $$
$$ 0 = \frac{x}{2 \sqrt{25 + x^2}} - \frac{1}{4} $$

Dado que hablamos de tiempo y por las velocidades que se tienen y las distancias, podemos pensar en que la distancia total podrá der hasta el triple de el lado conocido (5), si esto es de 15 millas y dividimos la distancia en 2 temos 7.5 millas a 2 millas por hora, son 3.75 horas (3 horas 40 minutos) y 7.5 millas por hora a 4 millas por hora que son 1.875 horas, un aproximado de 1 hora 40 minutos, un aproximado de 5 horas, así que partiremos de pensar que el menor tiempo posible son 2 horas y el máximo 8.

In [2]:
from math import sqrt

from metodos_numericos import metodo_de_secante
from utils import imprimir_tabla

def f(x):
    return (x / (2 * sqrt(25 + x**2))) - 0.25


resultado = metodo_de_secante(2, 8, f, tolerancia=1E-7, resultado=[])
imprimir_tabla(resultado)

┏━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┓
┃ # de iteración ┃ x0                ┃ x1                ┃ x2                ┃ error             ┃
┡━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━┩
│ 1              │ 2.000000000000000 │ 8.000000000000000 │ 3.619059152868730 │ 4.380940847131270 │
│ 2              │ 8.000000000000000 │ 3.619059152868730 │ 2.173547702928416 │ 1.445511449940314 │
│ 3              │ 3.619059152868730 │ 2.173547702928416 │ 2.954048401658996 │ 0.780500698730580 │
│ 4              │ 2.173547702928416 │ 2.954048401658996 │ 2.892557871216257 │ 0.061490530442738 │
│ 5              │ 2.954048401658996 │ 2.892557871216257 │ 2.886700210850719 │ 0.005857660365538 │
│ 6              │ 2.892557871216257 │ 2.886700210850719 │ 2.886751384541230 │ 0.000051173690511 │
│ 7              │ 2.886700210850719 │ 2.886751384541230 │ 2.886751345948385 │ <-- solución      │
└────────────────┴───────────────────┴───────────────────┴───────────────────┴───────────────────┘

Por lo tanto, la distancia a la que tiene que trazar su curso el navegante es a aproximadamente **2.886751345948385** millas del punto A.